internal/blueprint/customizations.go

```diff
@@ -155,11 +155,21 @@ func (c *Customizations) GetGroups() []GroupCustomization {
 }
 
 func (c *Customizations) GetKernel() *KernelCustomization {
-	if c == nil {
-		return nil
+	var name string
+	var append string
+	if c != nil && c.Kernel != nil {
+		name = c.Kernel.Name
+		append = c.Kernel.Append
 	}
 
-	return c.Kernel
+	if name == "" {
+		name = "kernel"
```

Comment:
Just a thing to remember, if we were to use that from the sysconfig stage in the future, the default there for fedora is `kernel-core`:

```
# DEFAULTKERNEL specifies the default kernel package type
DEFAULTKERNEL=kernel-core
```

Reply:
Ah, thank for pointing that out. I wonder if it matters / what difference it makes. It seems like `kernel` and `kernel-core` go together. Is this just another random difference between RHEL and Fedora?

Reply:
Good question, need to investigate this.

Reply:
I think it is a by-product of the auto-detection mechanism in anaconda: https://github.com/rhinstaller/anaconda/blob/6bbc4f12b1a3e61790ae3d742459714a82eb7c0b/pyanaconda/modules/storage/bootloader/utils.py#L149

Reply:
On Fedora it is auto-generated via anaconda, which detects the package by looking which package provides the actual kernel image (`kernel-core`, which is also true on RHEL). But on RHEL the `/etc/sysconfig/kernel` file for the guest images is explicitly written in the [kickstart](http://git.app.eng.bos.redhat.com/git/spin-kickstarts.git/tree/rhel8/rhel-8.4-kvm-x86_64.ks#n216) file and there set to `DEFAULTKERNEL=kernel`.

Reply:
Ok, I think this is only used by the [`new-kernel-pkg` script](https://github.com/rhboot/grubby/blob/99d10a31cee89eb45ac0743f484018d36500ce66/new-kernel-pkg#L846) and there both it checks for `$package` and `$package-core`, so both `kernel` and `kernel-core` will work. `new-kernel-pkg` is in `grubby-deprecated` and I think not used anymore on systems using BLS. Right @martinezjavier ?

Reply:
> . Right @martinezjavier ?

@gicmo yes, that's correct. The `DEFAULTKERNEL` option is `/etc/sysconfig/kernel` is not used anymore in neither Fedora nor RHEL8.